### ISSUE TITLE
Dynamic block ignores block schema

### ISSUE THREAD

**Issue by wata727** [terraform-linters/tflint#1576]
## Introduction

TFLint implicitly resolves dynamic blocks. For instance, it will return 2 blocks for the following configuration:

```hcl
resource "aws_instance" "dynamic" {
  dynamic "ebs_block_device" {
    for_each = var.block_devices
    content {
      encrypted = false
    }
  }

  ebs_block_device {
    encrypted = false
  }
}
```

```go
resources, _ := runner.GetResourceContent("aws_instance", &hclext.BodySchema{
	Blocks: []hclext.BlockSchema{
		{
			Type: "ebs_block_device",
			Body: &hclext.BodySchema{Attributes: []hclext.AttributeSchema{{Name: "encrypted"}}},
		},
	},
}, nil)

len(resources.Blocks[0].Body.Blocks) // => 2
```

But this dynamic block ignores the block type "ebs_block_device" and all dynamic blocks are returned.

## Expected Behavior

It should return 2 blocks for the following configuration like

```hcl
resource "aws_instance" "dynamic" {
  dynamic "network_interface" { # => should be ignored because this is not "ebs_block_device" block
    for_each = var.network_instances
    content {
      device_index = 0
    }
  }

  dynamic "ebs_block_device" {
    for_each = var.block_devices
    content {
      encrypted = false
    }
  }

  ebs_block_device {
    encrypted = false
  }
}
```

## Actual behavior

3 blocks are returned.

```go
resources, _ := runner.GetResourceContent("aws_instance", &hclext.BodySchema{
	Blocks: []hclext.BlockSchema{
		{
			Type: "ebs_block_device",
			Body: &hclext.BodySchema{Attributes: []hclext.AttributeSchema{{Name: "encrypted"}}},
		},
	},
}, nil)

len(resources.Blocks[0].Body.Blocks) // => 3

for _, block := range resources.Blocks[0].Body.Blocks {
	fmt.Printf("type: %s", block.Type)
}
// => type: network_interface
//    type: ebs_block_device
//    type: ebs_block_device
```

## Step to Reproduce

See also https://github.com/terraform-linters/tflint-ruleset-aws/issues/391

## Additional Context

```console
$ tflint -v
TFLint version 0.42.1
+ ruleset.aws (0.18.0)
+ ruleset.terraform (0.2.0-bundled)
$ terraform -v
Terraform v1.1.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.36.1
```
